e2e_playwright/st_plotly_chart_select.py

```diff
@@ -0,0 +1,189 @@
+
+st.header("Bar Chart with Click")
+data_canada = px.data.gapminder().query("country == 'Canada'")
+fig_bar = px.bar(data_canada, x="year", y="pop")
+st.plotly_chart(fig_bar, on_select="rerun", key="bar_chart", selection_mode=["points"])
+if len(st.session_state.bar_chart.select["points"]) > 0:
+    st.write("The original df data selected:")
+    points = st.session_state.bar_chart.select["points"]
+    # Extract x and y values directly into lists
+    x_values = [point["x"] for point in points]
+    y_values = [point["y"] for point in points]
+
+    # Use these lists to filter the DataFrame
+    filtered_df = data_canada[
+        data_canada["year"].isin(x_values) & data_canada["pop"].isin(y_values)
+    ]
+    st.dataframe(filtered_df)
+else:
+    st.write("Nothing is selected")
+
+
+st.header("Box Selections for a Stacked Bar Chart")
+wide_df = px.data.medals_wide()
+fig = px.bar(
+    wide_df, x="nation", y=["gold", "silver", "bronze"], title="Wide-Form Input"
+)
+fig.update_layout(dragmode="select")
+st.plotly_chart(
+    fig,
+    on_select="rerun",
+    key="StackedBar_chart",
+)
+if len(st.session_state.StackedBar_chart.select["points"]) > 0:
+    st.write("Countries and their medal data that were selected:")
+    points = st.session_state.StackedBar_chart.select["points"]
+    # Extract x and y values directly into lists
+    x_values = [point["x"] for point in points]
+
+    # Use these lists to filter the DataFrame
+    filtered_df = wide_df[wide_df["nation"].isin(x_values)]
+    st.dataframe(filtered_df)
+else:
+    st.write("Nothing is selected")
+
+# TODO(willhuang1997): Readd choropleth charts
+# st.header("Selections on Choropleth Chart with a callback")
+
+
+# @st.cache_data
+# def load_json(url):
+#     with urlopen(url) as response:
+#         counties = json.load(response)
+#         return counties
+
+
+# @st.cache_data
+# def load_data(url):
+#     df = pd.read_csv(url, dtype={"fips": str})
+#     return df
+
+
+# df = load_data(
+#     "https://raw.githubusercontent.com/plotly/datasets/master/fips-unemp-16.csv"
+# )
+# counties = load_json(
+#     "https://raw.githubusercontent.com/plotly/datasets/master/geojson-counties-fips.json"
+# )
+# fig = px.choropleth_mapbox(
+#     df,
+#     geojson=counties,
+#     locations="fips",
+#     color="unemp",
+#     color_continuous_scale="Viridis",
+#     range_color=(0, 12),
+#     mapbox_style="carto-positron",
+#     zoom=3,
+#     center={"lat": 37.0902, "lon": -95.7129},
+#     opacity=0.5,
+#     labels={"unemp": "unemployment rate"},
+# )
+# fig.update_layout(margin={"r": 0, "t": 0, "l": 0, "b": 0})
+# return_value = st.plotly_chart(
+#     fig,
+#     on_select="rerun",
+#     key="Choropleth_chart",
+# )
+# st.write("Data selected:")
+# if return_value:
+#     st.dataframe(return_value.select["points"])
+
+st.header("Lasso selections on Histograms with a callback")
+df = px.data.tips()
+fig = px.histogram(df, x="total_bill")
+
+
+def histogram_callback():
+    try:
```

Comment:
Instead of try & except here, it would probably be better to use the same if check as in the tests above.